core/src/main/scala/org/apache/spark/ui/jobs/StageTable.scala

```diff
@@ -150,6 +152,10 @@ private[ui] class StageTableBase(
     }
     val formattedDuration = duration.map(d => UIUtils.formatDuration(d)).getOrElse("Unknown")
 
+    val maxTaskDuration = stageData.taskData.values.toSeq
+        .map(d => d.taskInfo.duration).max
```

Comment:
OK, I'll report the maximum of finished and unfinished tasks using logic similar to that of the stage duration calculation.